File: GamesTournamentsWeb.Web/ClientApp/models/tournaments/TournamentOverview.ts
import type { Game } from '~/models/games/Game'
import type { Platform } from '~/models/tournaments/Platform'
import type { Region } from '~/models/tournaments/Region'

interface TournamentOverview {
    id: number
    name: string
    teamSize: number
    game: Game
    platform: Platform
    regions: Region[]
    startDate: Date
    endDate: Date
}

export type { TournamentOverview }

File: GamesTournamentsWeb.DataAccess/Models/Tournaments/TournamentOverview.cs
using GamesTournamentsWeb.DataAccess.Models.Games;

namespace GamesTournamentsWeb.DataAccess.Models.Tournaments;

public class TournamentOverview
{
    public int Id { get; set; }
    public string Name { get; set; }    
    public int TeamSize { get; set; }
    public int GameId { get; set; }
    public Game Game { get; set; }
    public int PlatformId { get; set; }
    public Platform Platform { get; set; }
    public ICollection<Region> Regions { get; set; }
    public DateTimeOffset StartDate { get; set; }
    public DateTimeOffset? EndDate { get; set; }
}
